ext/liquid_c/block.h

```diff
@@ -1,17 +1,26 @@
 #if !defined(LIQUID_BLOCK_H)
 #define LIQUID_BLOCK_H
 
-#include "vm_assembler.h"
-
 typedef struct block_body {
-    VALUE obj;
-    vm_assembler_t code;
-    VALUE source; // hold a reference to the ruby object that OP_WRITE_RAW points to
+    size_t instructions_offset;
+    size_t instructions_bytes;
```

Comment:
It seems like this is the serialized form for the block body, in which case we want to avoid using `size_t` values, since that seems like unnecessary coupling to the CPU architecture that it is compiled on.  If we only support a 24-bit length source string, then we can safely use `uint32_t` for offsets and lengths into the serialized code, assuming a byte of source can't compile into more than 256 bytes of compiled code.

Reply:
I think we should use `uint64_t` for the offset because since we are writing every block body to this shared buffer, the buffer size could overflow a `uint32_t` (e.g. if each file compiled `2^24` bytes of instructions and we have `2^8=256` files).

Reply:
A document body would map to only a single file.  Each partial would have its own document body.  However, if we did decide to combine them in the future, then we would have to worry more about overflows.